src/coreclr/vm/amd64/JitHelpers_Slow.asm

```diff
@@ -197,69 +195,6 @@ LEAF_ENTRY JIT_TrialAllocSFastSP, _TEXT
         jmp     JIT_NEW
 LEAF_END JIT_TrialAllocSFastSP, _TEXT
 
-; HCIMPL2(Object*, JIT_Box, CORINFO_CLASS_HANDLE type, void* unboxedData)
-NESTED_ENTRY JIT_BoxFastUP, _TEXT
-
-        ; m_BaseSize is guaranteed to be a multiple of 8.
-        mov     r8d, [rcx + OFFSET__MethodTable__m_BaseSize]
-
-        inc     [g_global_alloc_lock]
-        jnz     JIT_Box
-
-        mov     rax, [g_global_alloc_context + OFFSETOF__ee_alloc_context__alloc_ptr]       ; alloc_ptr
-        mov     r10, [g_global_alloc_context + OFFSETOF__ee_alloc_context__m_CombinedLimit] ; m_CombinedLimit
-
-        add     r8, rax
-
-        cmp     r8, r10
-        ja      NoAlloc
-
-        test    rdx, rdx
-        je      NullRef
-
-        mov     qword ptr [g_global_alloc_context + OFFSETOF__ee_alloc_context__alloc_ptr], r8 ; update the alloc ptr
-        mov     [rax], rcx
-        mov     [g_global_alloc_lock], -1
-
-        ; Check whether the object contains pointers
-        test    dword ptr [rcx + OFFSETOF__MethodTable__m_dwFlags], MethodTable__enum_flag_ContainsGCPointers
-        jnz     ContainsPointers
-
-        ; We have no pointers - emit a simple inline copy loop
-
-        mov     ecx, [rcx + OFFSET__MethodTable__m_BaseSize]
-        sub     ecx, 18h  ; sizeof(ObjHeader) + sizeof(Object) + last slot
-
-    CopyLoop:
-        mov     r8, [rdx+rcx]
```

Comment:
This fast copy loop had a subtle bug. It will crash if the value type to box is at the end of the page and it does not have size that is multiple of pointer size. 

(Just pointing this out.)